### PR TITLE
Set up new logging artifact for SymNode

### LINKED ISSUE #113876: Fine grained SymNode logging
### 🐛 Describe the bug

I recently found these logs helpful

```
diff --git a/torch/fx/experimental/sym_node.py b/torch/fx/experimental/sym_node.py
index 47ce9b9d3d5..8a7cc0bbf1b 100644
--- a/torch/fx/experimental/sym_node.py
+++ b/torch/fx/experimental/sym_node.py
@@ -817,6 +817,7 @@ def _make_node_magic(method, func):
             log.warning("failed to eval %s(%s, %s)", method, self.expr, other.expr)
             raise
         out = safe_expand(out)
+        log.debug("%s %s %s -> %s", func, self.expr, other.expr, out)
         pytype: Type
         # This is not strictly correct. In Python, a**b may return complex when
         # a < 0 and b is a float: (-1)**2.1. Same for sympy.sqrt(-3.14). This
@@ -857,6 +858,7 @@ def _make_node_magic(method, func):
         except Exception:
             log.warning("failed to eval %s(%s)", method, expr)
             raise
+        log.debug("%s %s -> %s", func, expr, out)
 
         out_hint = None
         if self.hint is not None:
@@ -1045,6 +1047,7 @@ def _make_user_magic(method, user_type):
         return wrap_node(getattr(self.node, method_attr)())
 
     def binary_magic_impl(self, other):
+        log.debug("MAGIC %s %s %s", method, self, other)
         if is_constant(self):
             return (method_to_operator(method))(get_constant(self), other)
         if is_constant(other):
```

Probably should get its own artifact though.

### Versions

main

cc @msaroufim @wconstab @bdhirsh @anijain2305 @zou3519

### PR DESCRIPTION
**Pull request by PrincipalsOffice**
Fixes #113876

Hi, I updated various logging configs and the SymNode module to use the new dedicated logging artifact. This is my first pytorch PR, mirrored my changes off of https://github.com/pytorch/pytorch/pull/111808.

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @chenyang78 @aakhundov @kadeng @ezyang

### CODE REVIEW

File: torch/fx/experimental/sym_node.py [hunk 1]
```diff
@@ -1097,7 +1096,7 @@ def sizes_strides_impl(self, sizes, strides):
         try:
             out = func(size_exprs, stride_exprs)
         except Exception:
-            log.warning("failed to eval %s(%s, %s)", method, size_exprs, stride_exprs)
+            sym_node_log.warning("failed to eval %s(%s, %s)", method, size_exprs, stride_exprs)
```
Comment (ezyang) [inline torch/fx/experimental/sym_node.py:1099]: Would you kindly revert the changes for the warnings? The warnings should be emitted by default with or without the sym_node artifact being enabled.

**Comment by ezyang** [comment]
Looks good, thanks.

**Comment by PrincipalsOffice** [comment]
Makes sense, I've reverted those warning log changes.

**Review by ezyang — APPROVED** [review]
